framework/include/utils/TimedPrint.h

```diff
@@ -79,6 +81,10 @@ class TimedPrint final
 #define ARGS args_in...
 
 #endif
+    if (_active_instance)
```

Comment:
This isn't thread safe. You probably should protect access to _active_instance with a mutex.  Probably want to try_lock here.

Reply:
It doesn't need to be thread safe. We don't ever print, nor would you ever want to try to print in a threaded region. These instantiations always appear in serial regions. The mutex would be an overkill for console outputting.

Reply:
Up to you - but I bet sometime in our future somebody will spend half a day figuring out why their timed print is doing something wonky.

Reply:
I expect that it'll lock, which is probably a good thing. Let me try it out! 

How about I put in some explicit documentation that this object should not be used in a threaded region, nor would you want it to be?

Reply:
I expect that it'll lock, which is probably a good thing. Let me try it out!

How about I put in some explicit documentation that this object should not be used in a threaded region?

Reply:
Oh - one more thing, this object isn't included in any of the pluggable system base classes so it's not even available. It's being used heavily in FEProblem and MooseApp, which are mostly controlled by us.

Reply:
Two threads could do+pass the if check simultaneously and both proceed to stream stuff.  It is unlikely but possible.  Sure - docs are fine.  But a mutex would be shorter to write than the docs :-)

Reply:
The purpose of this class is to print progress information to the console for long running operations. I'm not aware of any places where we print from threaded regions anywhere in the framework and that will never change, let alone the need to use a class that prints progress information because the threaded operation takes a long time. It just doesn't happen. 

Normally I'm the one that likes to protect against silly mistakes but I can't even come up with a single use case that's even close to needing this object in a threaded region.

Reply:
Well - when the day comes - I hope you'll be happy.

Reply:
OK - I just attempted to create a test case where I stuck this macro in a threaded region (e.g. computeQpResidual). To increase the likelihood of running into a thread issue. I put in a random sleep for a few milliseconds using the `std::chrono` stuff and ran with tens to hundreds of threads. I never ran into an issue. I thought about it more and realized that the macro instantiates an object in the current stack frame and uses RAII to control the termination and cleanup of the printing operation. With the way it's coded you are going to always use it in that manner (e.g. the current stack frame of the current thread). It's not really designed for declaring and using later (printing starts in the constructor and is terminated in the destructor). So for all uses of this method you are going to end up with a separate instance per thread if you attempted to put it in a threaded region.

Reply:
Yes but each instance refers to the same static member.  It is unlikely, but possible.  Both threads would have to complete the if check before any other threads set the static var to be non-null.